daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -1350,6 +1350,8 @@ private[validation] object Typing {
         typeOfCreateInterface(iface, arg)
       case UpdateExercise(tpl, choice, cid, arg) =>
         typeOfExercise(tpl, choice, cid, arg)
+      case UpdateSoftExercise(tpl, choice, cid, arg) =>
+        typeOfExercise(tpl, choice, cid, arg)
```

Comment:
as in the Haskell typechecker, we want to prevent `softExercise` from being used on templates with keys, at least for now.

```suggestion
        typeOfSoftExercise(tpl, choice, cid, arg)
```

Reply:
Shall I leave this for the future PR which deals with typing issues?

Reply:
I added a 16151 todo